### PR TITLE
RFC: Sm redesign

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/include/sm/sm.h [hunk 1]
```diff
@@ -30,65 +31,83 @@
 
 #include <types_ext.h>
 
-struct sm_nsec_ctx {
```
Comment (etienne-lms) [inline core/arch/arm/include/sm/sm.h]: from `sm_save_modes_regs`, i think the registers are stored in opposite u32 order.
not an issue as all regs manipulated by 'sm' are outside `sm_mode_regs` structure.

Comment (jenswi-linaro) [inline core/arch/arm/include/sm/sm.h]: Yes, you're right. I'll fix.

**Comment by etienne-lms** [comment]
looks ok to me. 1 more required change: remove `#include <sm_defs.h>` from `plat-sunxi/platform.c` (see travis build error on plat-sunxi).

**Comment by jenswi-linaro** [comment]
Travis fails on a missing Singed-off-by on one of the "[review]" patches.
If we're happy with this I'd like to squash and rebase on master for some final testing on some real hardware (preferably both STM and TI).

**Comment by etienne-lms** [comment]
i am happy with these changes.
i'll test them on my STM board and tell you (once i've fixed my env that is currently corrupted...).

**Comment by jenswi-linaro** [comment]
Squashed and rebased on master.
Please test on real hardware.
@glneo, would you mind testing on plat-ti?

**Comment by jbech-linaro** [comment]
`Tested-by: Joakim Bech <joakim.bech@linaro.org>` (RPi3)

(with https://github.com/linaro-swg/linux/pull/28 applied).

**Comment by glneo** [comment]
Works well on my side,

Tested-by: Andrew F. Davis <afd@ti.com> (plat-ti)

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org> (stm-b2260)`

**Comment by jenswi-linaro** [comment]
Tags applied

**Comment by etienne-lms** [comment]
for info, i've just tested this with TEE pager enable and faced boot
issues. (sorry i did not checked before)
seems the secondary cores stack is corrupted in some way
i will investigate more and give feedback...

**Pull request merged**